src/main/java/org/ieeervce/gatekeeper/config/UserInfoUserDetailsService.java

```diff
@@ -0,0 +1,21 @@
+package org.ieeervce.gatekeeper.config;
+
+import org.ieeervce.gatekeeper.entity.User;
+import org.ieeervce.gatekeeper.repository.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+
+import java.util.Optional;
+
+public class UserInfoUserDetailsService implements UserDetailsService {
```

Comment:
Name is a bit confusing.

I would suggest an alternative, but don't have much better ideas myself.

(For now: add a JSDoc comment explaining what it does)